gravitee-am-identityprovider/gravitee-am-identityprovider-common/src/main/java/io/gravitee/am/identityprovider/common/oauth2/authentication/AbstractOpenIDConnectAuthenticationProvider.java

```diff
@@ -194,7 +233,12 @@ protected Maybe<Token> authenticate(Authentication authentication) {
         urlParameters.add(new BasicNameValuePair(Parameters.CLIENT_ID, getConfiguration().getClientId()));
         urlParameters.add(new BasicNameValuePair(Parameters.REDIRECT_URI, String.valueOf(authentication.getContext().get(Parameters.REDIRECT_URI))));
         urlParameters.add(new BasicNameValuePair(Parameters.CODE, authorizationCode));
-        urlParameters.add(new BasicNameValuePair(Parameters.GRANT_TYPE, "authorization_code"));
+        urlParameters.add(new BasicNameValuePair(Parameters.GRANT_TYPE, GrantType.AUTHORIZATION_CODE));
+        if (getConfiguration().usePkce()) {
+            Optional.ofNullable((String) authentication.getContext().get(ConstantKeys.IDP_CODE_VERIFIER))
+                    .ifPresent(codeVerifier -> urlParameters.add(new BasicNameValuePair(Parameters.CODE_VERIFIER, codeVerifier)));
```

Comment:
**Question**: do we have to do something if the codeverifier is missing when PKCE is enabled :thinking: We may have to throw an error... I'm not sure... but we can at least log a WARN wdyt ?

Reply:
Not sure about an error, technically there is a somehwat legal scenario where we have PKCE enabled but the code verifier won't be available here:

1. Social IDP  gets configured, with PKCE disabled
2. user goes to /authorize of our app, gets redirected to /login. Social IDP's sign-in URL is generated without code challenge because PKCE is disabled
3. admin enables PKCE for the IDP
4. user initializes login via the social IDP - there's no code challenge, so also no code verifier

Reply:
I've added a warning, so that we can see whether this situation is going to happer